.pre-commit-config.yaml

```diff
@@ -0,0 +1,97 @@
+ci:
+  autofix_prs: false
+  autoupdate_schedule: monthly
+
+repos:
+
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.4.0
+  hooks:
+  - id: check-ast
+  - id: check-merge-conflict
+    exclude: .*\.rst
+  - id: check-case-conflict
+  - id: name-tests-test
+    args: [--pytest-test-first]
+  - id: trailing-whitespace
+  - id: end-of-file-fixer
+  - id: check-json
+  - id: check-toml
+  - id: check-yaml
+
+- repo: https://github.com/sirosen/texthooks
+  rev: 0.5.0
+  hooks:
+  - id: fix-smartquotes
+  - id: fix-spaces
+  - id: fix-ligatures
+  - id: forbid-bidi-controls
+
+- repo: https://github.com/Lucas-C/pre-commit-hooks
+  rev: v1.5.4
+  hooks:
+  - id: remove-crlf
+
+- repo: https://github.com/macisamuele/language-formatters-pre-commit-hooks
+  rev: v2.10.0
+  hooks:
+  - id: pretty-format-ini
+    args: [--autofix]
+
+- repo: https://github.com/MarcoGorelli/absolufy-imports
+  rev: v0.3.1
+  hooks:
+  - id: absolufy-imports
+    exclude: docs/plasmapy_sphinx
+
+- repo: https://github.com/pre-commit/pygrep-hooks
+  rev: v1.10.0
+  hooks:
+  - id: python-check-blanket-noqa
+  - id: rst-directive-colons
+  - id: rst-inline-touching-normal
+  - id: text-unicode-replacement-char
+
+- repo: https://github.com/codespell-project/codespell
+  rev: v2.2.5
+  hooks:
+  - id: codespell
+    args: [--write-changes]
+    additional_dependencies:
+    - tomli
+
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.0.287
+  hooks:
+  - id: ruff
+    args: [--fix]
+
+- repo: https://github.com/PyCQA/isort
+  rev: 5.12.0
+  hooks:
+  - id: isort
+    name: isort
+    entry: isort
+    require_serial: true
+    language: python
+    types:
+    - python
+
+- repo: https://github.com/psf/black
+  rev: 23.9.0
+  hooks:
+  - id: black
+
+- repo: https://github.com/asottile/blacken-docs
+  rev: 1.16.0
+  hooks:
+  - id: blacken-docs
+
+- repo: https://github.com/nbQA-dev/nbQA
+  rev: 1.7.0
+  hooks:
+  - id: nbqa-check-ast
+  - id: nbqa-ruff
+    args: ['--select=A,ARG,B,BLE,C,C4,E,F,FLY,I002,INT,ISC,PERF,PIE,PLC,PLE,PYI,Q003,RET,RSE,SIM,TID,TRY,UP,W', '--ignore=B007,B008,B015,B018,E402,E501,E741,SIM114']
+  - id: nbqa-isort
+  - id: nbqa-black
```

Comment:
This is for applying linters like black, isort, and ruff to example notebooks.  If we don't plan to add example notebooks, then this can be removed.